src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -109,4 +114,13 @@ private static void setCheckProperties(InputConfiguration.Builder inputConfigBui
             }
         }
     }
+
+    private static void setViolationLineNumbers(InputConfiguration.Builder inputConfigBuilder,
+                                    List<String> lines) {
+        for (int lineNo = 2; lineNo < lines.size(); lineNo++) {
+            if (VIOLATION_PATTERN.matcher(lines.get(lineNo)).find()) {
```

Comment:
i think it should be `matches` instead of `find`, 
as far as I understand `find` is used for groups, please correct me if I am wrong

Reply:
```matches``` is used for matching the whole text.
```find``` is used for matching a substring.

if we use ```find``` we should modify the regexp from ```.*//\s*violation(?:$|\W+.*$)``` to ```//\s*violation(?:$|\W+.*$)```

Reply:
> we should modify the regexp from

why? it should work without modifications

Reply:
yes but it will match the entire line. so using ```find``` will be pointless.

Reply:
can you rewrite to `matches`, lets see

Reply:
dine